### PR TITLE
Adjust merchant_reference checks

### PR DESCRIPTION
**Pull request by andersem**
We will sometimes get random errors when trying to update merchant_reference_2 on the transaction.

Trying to adjust the behaviour a little bit:

1) If the call fails, retry once
2) If it still fails, don't fail the order, but keep it open
3) When capturing, don't check that the merchant_reference is equal to the order_id, as we already can be sure that the transaction id is correct

### CODE REVIEW

**Comment by frodeaa** [comment]
Why are the update failing?

File: includes/class-wc-dintero-hp-ajax.php [hunk 1]
```diff
@@ -575,7 +575,16 @@ public function dhp_create_order() {
...
 					$fail_reason = __( 'Failed updating transaction with order_id ' ) . '.';
-					self::failed_order( $order, $transaction_id, $fail_reason );
+					$order->add_order_note( $fail_reason );
+
+					$update_response_retry = self::_adapter()->update_transaction(
+						$transaction_id, $update_payload
+					);
+
+					if (is_wp_error($update_response_retry)) {
+						$fail_reason = __( 'Failed updating transaction with order_id. Will stop trying ' ) . '.';
```
Comment (frodeaa) [inline includes/class-wc-dintero-hp-ajax.php:585]: Do we need to add another note if success?

File: includes/class-wc-gateway-dintero-hp.php [hunk 2]
```diff
@@ -797,7 +797,14 @@ public function process_payment_handler( $order_id ) {
 					self::failed_order( $order, $dintero_order_transaction_id, $fail_reason );
 				} else {
 					$fail_reason = __( 'Failed updating transaction with order_id ' ) . '.';
-					self::failed_order( $order, $dintero_order_transaction_id, $fail_reason );
+					$order->add_order_note( $fail_reason );
+
+					$update_response_retry = WCDHP()->checkout()->update_transaction($dintero_order_transaction_id, $order_id);
+
+					if (is_wp_error($update_response_retry)) {
+						$fail_reason = __( 'Failed updating transaction with order_id. Will stop trying ' ) . '.';
```
Comment (frodeaa) [inline includes/class-wc-gateway-dintero-hp.php:805]: maybe state the effects of this error? i.e. finding the order in the settlement?

**Review by frodeaa — APPROVED** [review]

File: includes/class-wc-dintero-hp-ajax.php (hunk 1 above)
Comment (andersem) [reply includes/class-wc-dintero-hp-ajax.php:585]: Added

File: includes/class-wc-gateway-dintero-hp.php (hunk 2 above)
Comment (andersem) [reply includes/class-wc-gateway-dintero-hp.php:805]: Yup

**Pull request merged**